File: 3471-MinimumAverageOfSmallestAndLargestElements/3471-MinimumAverageOfSmallestAndLargestElements.cpp
// Last updated: 7/17/2026, 4:23:51 PM
class Solution {
public:
    double minimumAverage(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        int i = 0;
        int j = nums.size() - 1;
        double minAvg = INT_MAX;

        while(i < j){
            double avg = (nums[i] + nums[j]) / 2.0;
            minAvg = min(minAvg, avg);
            i++;
            j--;
        }

        return minAvg;
    }
};

File: 3471-MinimumAverageOfSmallestAndLargestElements/3471-MinimumAverageOfSmallestAndLargestElements.js
// Last updated: 7/17/2026, 4:23:52 PM
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumAverage = function(nums) {
    nums = nums.sort((a,b)=>a-b);

    let left = 0;
    let right = nums.length-1;
    let minAvg = +Infinity;

    while(left < right){
        let avg = (nums[left]+nums[right]) / 2;
        minAvg = Math.min(minAvg , avg);
        left++;
        right--;
    }
    return minAvg
};